x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/query/CurationQueryRewriter.java

```diff
@@ -39,12 +42,18 @@ public QueryBuilder rewriteQuery(
         RelevanceMatchQueryBuilder relevanceMatchQuery,
         SearchExecutionContext context
     ) {
-        CurationSettings curationSettings = getSettings(relevanceMatchQuery, context);
+        CurationsGroup curationGroup = getSettings(relevanceMatchQuery, context);
+
+        if (curationGroup == null) {
+            return baseQuery;
+        }
 
-        if (curationSettings == null || curationSettings.conditions().stream().noneMatch(c -> c.match(relevanceMatchQuery))) {
+        final Optional<CurationSettings> optionalCurationSettings = curationGroup.findMatching(relevanceMatchQuery);
```

Comment:
It's interesting to think about whether we'll want to specify a default `CurationSettings` if one isn't found for the given `relevanceMatchQuery`. I think that's outside the scope of this PR but worth thinking about.

Reply:
Sure! I think the beauty of having services is that we can just return a default implementation. We can think about it when the time comes 👍